src/textual/_node_list.py

```diff
@@ -13,7 +14,7 @@ class DuplicateIds(Exception):
 
 
 @rich.repr.auto(angular=True)
-class NodeList(Sequence):
+class NodeList(Sequence[Widget]):
```

Comment:
Sequence is Generic and needs a type

Reply:
EDIT: importing `Widget` here causes cyclic dependency. Basically it makes DOM depend on Widget, and Widget itself obviously depends on DOM.

I resolved this by using `T` instead of `Widget` here. Apparently there's nothing in the code that needs an actual widget, only hashable things.

Reply:
Try putting Widget in quotes.

Reply:
I can do that, but isn't what I did now better? It really doesn't need to be Widget at all.

Reply:
The object is intended to be a container of Widgets. Better to type it as such.

Reply:
well in that case there's an actual cyclic dependency, no? I may be able to solve with quotes etc, but regardless there's a sort of "smell" to it. Why would DOMNode know about widgets? Anyway I'll have a look-see.

Reply:
Done. Works. (at least: passes mypy --strict and the tests...)